data_pipeline.py

```diff
@@ -0,0 +1,264 @@
+    example['intrinsic_matrix'] = tf.io.parse_tensor(
+        example['intrinsic_matrix'], out_type=tf.float64)
+    example['axis_matrix'] = tf.io.parse_tensor(
+        example['axis_matrix'], out_type=tf.float32)
+    example['box_objects'] = tf.io.parse_tensor(
+        example['box_objects'], out_type=tf.float64)
+
+    # Linearly scales each image in images to have mean 0 and variance 1.
+    example['images'] = tf.map_fn(
+        lambda image: tf.image.per_image_standardization(image),
+        example['images'])
+
+    return example
+
+
+def get_images_with_boxes(scene_dir):
+    """Get selected random images that have most boxes
+    Args:
+        scene_dir: directory path for scene
+    Return: image paths, depth_paths and extrinsic paths
+    """
+
+    image_paths = os.path.join(scene_dir, 'color')
+    depth_paths = os.path.join(scene_dir, 'depth')
+    extrinsic_paths = os.path.join(scene_dir, 'pose')
+
+    scene = os.path.basename(scene_dir)
+    axis_matrix = get_axismatrix(os.path.join(scene_dir, scene + '.txt'))
+    intrinsic_matrix = np.loadtxt(
+        os.path.join(
+            scene_dir,
+            'intrinsic_color.txt'))
+    box_objects = np.loadtxt(os.path.join(scene_dir, 'box_coords.txt'))
+
+    # assuming all images have the same width and height, change if not true
+    height = 968
+    width = 1296
+    train_size = 15
+
+    images = os.listdir(image_paths)
+    center_counts = []
+    for image in images:
+        if not image.endswith('.jpg'):
+            center_counts.append(0)
+            continue
+        extrinsic_matrix = np.loadtxt(os.path.join(
+            extrinsic_paths, image[:-3] + 'txt'))
+        boxes = projection_to_rgb(
+            box_objects,
+            intrinsic_matrix,
+            extrinsic_matrix,
+            axis_matrix)
+
+        # check if center of box in image
+        count = 0
+        for i, b in enumerate(boxes):
+            b = np.float32(b)
+            x, y = b[:, 0], b[:, 1]
+            if (x[0] < width) and (y[0] < height):
+                count += 1
+        center_counts.append(count)
+
+    # sort in center counts in descending order and select the top 30 images
+    center_indexes = np.argsort(center_counts)[::-1][:30]
+    assert center_indexes.size >= train_size, "reduce train size"
+    # randomly select train_size images from the list
+    np.random.seed(2000)
+    center_indexes = np.random.choice(
+        center_indexes, train_size, replace=False)
+
+    image_paths = [os.path.join(image_paths, images[i])
+                   for i in center_indexes]
+    depth_paths = [os.path.join(depth_paths, images[i][:-3] + 'png')
+                   for i in center_indexes]
+    extrinsic_paths = [os.path.join(
+        extrinsic_paths, images[i][:-3] + 'txt') for i in center_indexes]
+
+    return image_paths, depth_paths, extrinsic_paths
+
+
+def get_scene_data(scenename):
+    """Reads input data from scene
```

Comment:
If you require any pre-processing to be done to the ScanNet prior to running this function, explain it here.